### ISSUE TITLE
[bitnami/zookeeper] Enable JSON format structured logging

### ISSUE THREAD

**Issue by janhoy** [bitnami/containers#51921]
### Name and Version

bitnami/zoookeeper:3.9.1

### What is the problem this feature will solve?

More and more cloud native users request JSON-format structured logging which is easier to aggregate and parse by log collectors. Enabling JSON logging in e.g. zookeeper helm chart today will require users to build their own custom Docker image with some additional JAR libs and a custom logback configuration file, which adds a burden on every single user. Thus adding this to the container project seems like a good fit.

Log output will look similar to this, one JSON log object per line:
```json
{"@timestamp":"2023-10-10T18:14:26.041Z","log.level": "INFO","message":"Reading configuration from: /Users/janhoy/git/zookeeper/zookeeper-assembly/target/apache-zookeeper-3.10.0-SNAPSHOT-bin/bin/../conf/zoo.cfg","ecs.version": "1.2.0","service.name":"zookeeper","service.version":"3.10.0-SNAPSHOT","service.node.name":"MacBook-Pro-7.local","event.dataset":"zookeeper","process.thread.name":"main","log.logger":"org.apache.zookeeper.server.quorum.QuorumPeerConfig"}
{"@timestamp":"2023-10-10T18:14:26.047Z","log.level": "INFO","message":"clientPortAddress is 0.0.0.0:2181","ecs.version": "1.2.0","service.name":"zookeeper","service.version":"3.10.0-SNAPSHOT","service.node.name":"MacBook-Pro-7.local","event.dataset":"zookeeper","process.thread.name":"main","log.logger":"org.apache.zookeeper.server.quorum.QuorumPeerConfig"}
```

### What is the feature you are proposing to solve the problem?

I'd like it to be easy to enable JSON logging in bitnami/zookeeper docker image. Ideally by simply adding `ZOO_LOG_FORMAT=JSON`. The default should still be normal text logging.

The needed changes to will include
* Add a `logback.xml` to the repo, with configuration for JSON logging (appender / encoder)
* Download and add two new jar libraries to `/opt/bitnami/zookeeper/lib`: `co.elastic.logging:logback-ecs-encoder:1.5.0` and `co.elastic.logging:ecs-logging-core:1.5.0` and optionally `org.codehaus.janino:janino:3.1.10`.
* Introduce a new env.var e.g. `ZOO_LOG_FORMAT=json` to enable the feature.

There are several competing JSON schemas for logging, but the above suggests choosing ECS, as [recently selected by the OpenTelemetry project](https://opentelemetry.io/blog/2023/ecs-otel-semconv-convergence/) as the standard logging schema. So this has a good chance of becoming industry standard.

I have a draft pull request to show what it would look like.

### What alternatives have you considered?

I have filed an issue in upstream Zookeeper project (https://issues.apache.org/jira/browse/ZOOKEEPER-4757). But the maintainers rejected it, wanting downstream packagers to handle everything around logging configuration.

Downloading the dependency jar files is a tricky part of this when done outside of the zookeeper maven build. One solution could have been to re-build zookeeper, but I settled instead on pulling the jar files and validating the sha-sum as part of building the dockerfile. For this I added a new script `libmaven.sh`, but perhaps Bitnami already has a utility script to pull jar files from maven repository?

**Comment by javsalgar** [comment]
Hi,

Thank you so much for the PR! The team will take a look

**Comment by janhoy** [comment]
I revived this by posting a new PR with sign-off, please re-open this issue.

**Comment by carrodher** [comment]
Thank you for submitting the associated Pull Request. Our team will review it and provide feedback. Once the PR is merged, the issue will automatically close.

Your contribution is greatly appreciated!

**Comment by janhoy** [comment]
The PR had a review comment but it is still in discussion phase, so please keep this issue open.
